codingStandards/CodingStandard-CSharp.md

```diff
@@ -0,0 +1,367 @@
+# C# Coding Standard
+
+**Important:** Use [CSharpGuidelines](https://github.com/dennisdoomen/CSharpGuidelines) for any topics not covered in this document.
+
+- [File Organization](#file-organization)
+- [Formatting](#formatting)
+ - [Layout](#layout)
+ - [Indentation](#indentation)
+ - [White Space](#white-space)
+- [Naming Conventions](#naming-conventions)
+- [Commenting Conventions](#commenting-conventions)
+- [Language Conventions](#language-conventions)
+- [References](#references)
+
+## File Organization
+**1. Source file name and class name should always match.**<br>
+MyClass.cs → `public class MyClass { ... }`
+
+**2. Directory names should follow namespace for the class.**<br>
+`System.Windows.Forms.Control` should use the path System\Windows\Forms\Control.cs
+> Do not use namespace name with dots. This will make it easier to map namespaces to the directory layout.
+
+## Formatting
+### Layout
+**1. Avoid putting multiple namespaces or classes per file.**<br>
```

Comment:
If you use headings (e.g. `####`), GitHub auto-generates anchors that can be used from other places to refer to a specific rule easily.